lib/rules/no-document-location.js

```diff
@@ -22,7 +22,15 @@ module.exports = {
     create(context) {
         return {
             'Program:exit'() {
-                const ref = context.getScope().through.find(({ identifier }) => {
+                // The top level scope is the "global" scope. Depending on the "env" property in the
+                // eslint config, references might not be resolved at the global scope. The most
+                // reliable way to get all the unresolved references from the module is to get them
+                // from the "module" scope.
+                const moduleScope = context
+                    .getScope()
+                    .childScopes.find((childScope) => childScope.type === 'module');
+
```

Comment:
Does this work regardless of `env` property ☝️ ?

Reply:
It should work regardless on the environment. The environment sets the variable in the `root` scope.
The `module` should contain all the unresolved variables.